From e4ed5a099fbafc6a8902d9fff731e91c338ccea2 Mon Sep 17 00:00:00 2001
From: David Rokhvarg <David.Rokhvarg@amd.com>
Date: Thu, 21 Jan 2016 15:52:43 -0500
Subject: [PATCH 0714/1110] drm/amd/dal: Removed dce_version from dc_context.

Also, improved valdation of 'number of links' to create,
which take into account 'Diagnostics' environment.

Signed-off-by: David Rokhvarg <David.Rokhvarg@amd.com>
Acked-by: Harry Wentland <harry.wentland@amd.com>
---
 drivers/gpu/drm/amd/dal/dc/core/dc.c        | 20 ++++++++++++--------
 drivers/gpu/drm/amd/dal/dc/core/dc_target.c | 10 +++++-----
 drivers/gpu/drm/amd/dal/include/dal_types.h |  2 --
 3 files changed, 17 insertions(+), 15 deletions(-)

diff --git a/drivers/gpu/drm/amd/dal/dc/core/dc.c b/drivers/gpu/drm/amd/dal/dc/core/dc.c
index 81bcc1e..d953784 100644
--- a/drivers/gpu/drm/amd/dal/dc/core/dc.c
+++ b/drivers/gpu/drm/amd/dal/dc/core/dc.c
@@ -83,17 +83,23 @@ static bool create_links(struct dc *dc, const struct dc_init_data *init_params)
 
 	dc->link_count = 0;
 
-	dcb = (struct dc_bios*)dal_adapter_service_get_bios_parser(init_params->adapter_srv);
+	dcb = dal_adapter_service_get_bios_parser(init_params->adapter_srv);
 
 	connectors_num = dcb->funcs->get_connectors_number(dcb);
 
-	if (0 == connectors_num || connectors_num > ENUM_ID_COUNT) {
-		dal_error("DC: Invalid number of connectors!\n");
+	if (connectors_num > ENUM_ID_COUNT) {
+		dal_error("DC: Number of connectors %d exceeds maximum of %d!\n",
+				connectors_num, ENUM_ID_COUNT);
 		return false;
 	}
 
-	dal_output_to_console("%s: connectors_num:%d\n", __func__,
-			connectors_num);
+	if (connectors_num == 0 && init_params->num_virtual_links == 0) {
+		dal_error("DC: Number of connectors can not be zero!\n");
+		return false;
+	}
+
+	dal_output_to_console("DC: %s: connectors_num: physical:%d, virtual:%d\n",
+			__func__, connectors_num, init_params->num_virtual_links);
 
 	for (i = 0; i < connectors_num; i++) {
 		struct link_init_data link_init_params = {0};
@@ -335,8 +341,6 @@ static bool construct(struct dc *dc, const struct dal_init_data *init_params)
 
 	dc->ctx = dc_init_data.ctx;
 
-	dc->ctx->dce_version = dal_adapter_service_get_dce_version(
-			dc_init_data.adapter_srv);
 	dc->ctx->dce_environment = dal_adapter_service_get_dce_environment(
 			dc_init_data.adapter_srv);
 
@@ -522,7 +526,7 @@ bool dc_commit_targets(
 	dal_logger_write(dc->ctx->logger,
 				LOG_MAJOR_INTERFACE_TRACE,
 				LOG_MINOR_COMPONENT_DC,
-				"%s: %d targets",
+				"%s: %d targets\n",
 				__func__,
 				target_count);
 
diff --git a/drivers/gpu/drm/amd/dal/dc/core/dc_target.c b/drivers/gpu/drm/amd/dal/dc/core/dc_target.c
index e7df2e2..a71034c 100644
--- a/drivers/gpu/drm/amd/dal/dc/core/dc_target.c
+++ b/drivers/gpu/drm/amd/dal/dc/core/dc_target.c
@@ -460,7 +460,7 @@ void dc_target_log(
 	dal_logger_write(dal_logger,
 			log_major,
 			log_minor,
-			"core_target 0x%x: surface_count=%d, stream_count=%d",
+			"core_target 0x%x: surface_count=%d, stream_count=%d\n",
 			core_target,
 			core_target->status.surface_count,
 			core_target->public.stream_count);
@@ -472,7 +472,7 @@ void dc_target_log(
 		dal_logger_write(dal_logger,
 			log_major,
 			log_minor,
-			"core_stream 0x%x: src: %d, %d, %d, %d; dst: %d, %d, %d, %d;",
+			"core_stream 0x%x: src: %d, %d, %d, %d; dst: %d, %d, %d, %d;\n",
 			core_stream,
 			core_stream->public.src.x,
 			core_stream->public.src.y,
@@ -485,20 +485,20 @@ void dc_target_log(
 		dal_logger_write(dal_logger,
 			log_major,
 			log_minor,
-			"\tpix_clk_khz: %d, h_total: %d, v_total: %d",
+			"\tpix_clk_khz: %d, h_total: %d, v_total: %d\n",
 			core_stream->public.timing.pix_clk_khz,
 			core_stream->public.timing.h_total,
 			core_stream->public.timing.v_total);
 		dal_logger_write(dal_logger,
 			log_major,
 			log_minor,
-			"\tsink name: %s, serial: %d",
+			"\tsink name: %s, serial: %d\n",
 			core_stream->sink->public.edid_caps.display_name,
 			core_stream->sink->public.edid_caps.serial_number);
 		dal_logger_write(dal_logger,
 			log_major,
 			log_minor,
-			"\tlink: %d",
+			"\tlink: %d\n",
 			core_stream->sink->link->public.link_index);
 	}
 }
diff --git a/drivers/gpu/drm/amd/dal/include/dal_types.h b/drivers/gpu/drm/amd/dal/include/dal_types.h
index 73c13c4..0e16ebb 100644
--- a/drivers/gpu/drm/amd/dal/include/dal_types.h
+++ b/drivers/gpu/drm/amd/dal/include/dal_types.h
@@ -179,8 +179,6 @@ struct dc_context {
 	struct dal_logger *logger;
 	void *cgs_device;
 
-	/* Diagnostics */
-	enum dce_version dce_version;
 	enum dce_environment dce_environment;
 };
 
-- 
2.7.4

